src/ProductConstructionService/ProductConstructionService.Api/Queue/JobsProcessor.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Extensions.Options;
+using ProductConstructionService.Api.Queue.Jobs;
+
+namespace ProductConstructionService.Api.Queue;
+
+public class JobsProcessor(
+    ILogger<JobsProcessor> logger,
+    IOptions<JobProcessorOptions> options,
+    JobsProcessorScopeManager scopeManager,
+    QueueServiceClient queueServiceClient)
+    : BackgroundService
+{
+    private readonly ILogger<JobsProcessor> _logger = logger;
+    private readonly IOptions<JobProcessorOptions> _options = options;
+    private readonly JobsProcessorScopeManager _scopeManager = scopeManager;
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        await ProcessJobsAsync(cancellationToken);
+    }
+
+    private async Task ProcessJobsAsync(CancellationToken cancellationToken)
+    {
+        QueueClient queueClient = queueServiceClient.GetQueueClient(_options.Value.JobQueueName);
+        _logger.LogInformation("Starting to process PCS jobs {queueName}", _options.Value.JobQueueName);
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            using (_scopeManager.BeginJobScopeWhenReady(cancellationToken))
+            {
+                try
+                {
+                    await ReadAndProcessJobAsync(queueClient, cancellationToken);
+                }
+                catch (Exception ex)
```

Comment:
We spoke about adding here a `catch` for `OperationCancelledException` so that we can actually ever quit

Reply:
We will exit, because of the while loop condition, but we don't to log that part I think

Reply:
And we're fine with `BeginJobScopeWhenReady` throwing too?

Reply:
I think that's a good place to throw from, before we create a scope for the new job, we check if the whole thing was cancelled

Reply:
It is, but it will blow up through here

Reply:
I added a `catch` for `OperationCancelledException` to just break out of the while loop